tensorflow_datasets/image/wider_face.py

```diff
@@ -0,0 +1,187 @@
+	Title = {WIDER FACE: A Face Detection Benchmark},
+	Year = {2016}}
+"""
+
+_DESCRIPTION = """
+WIDER FACE dataset is a face detection benchmark dataset, of which images are 
+selected from the publicly available WIDER dataset. We choose 32,203 images and 
+label 393,703 faces with a high degree of variability in scale, pose and 
+occlusion as depicted in the sample images. WIDER FACE dataset is organized 
+based on 61 event classes. For each event class, we randomly select 40%/10%/50% 
+data as training, validation and testing sets. We adopt the same evaluation 
+metric employed in the PASCAL VOC dataset. Similar to MALF and Caltech datasets,
+we do not release bounding box ground truth for the test images. Users are 
+required to submit final prediction files, which we shall proceed to evaluate.
+"""
+
+
+class WiderFace(tfds.core.GeneratorBasedBuilder):
+  """WIDER FACE Dataset."""
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  def _info(self):
+    features = {
+        'image': tfds.features.Image(encoding_format='jpeg'),
+        'image/filename': tfds.features.Text(),
+        'faces': tfds.features.Sequence({
+            'bbox': tfds.features.BBoxFeature(),
+            'blur': tf.uint8,
+            'expression': tf.bool,
+            'illumination': tf.bool,
+            'occlusion': tf.uint8,
+            'pose': tf.bool,
+            'invalid': tf.bool,
+        }),
+    }
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict(features),
+        urls=[_PROJECT_URL],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    extracted_dirs = dl_manager.download_and_extract({
+        'wider_train': _WIDER_TRAIN_URL,
+        'wider_val': _WIDER_VAL_URL,
+        'wider_test': _WIDER_TEST_URL,
+        'wider_annot': _WIDER_ANNOT_URL,
+    })
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                'split': 'train',
+                'extracted_dirs': extracted_dirs
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                'split': 'val',
+                'extracted_dirs': extracted_dirs
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={
+                'split': 'test',
+                'extracted_dirs': extracted_dirs
+            })
+    ]
+
+  def _generate_examples(self, split, extracted_dirs):
+    """Yields examples."""
+    pattern_fname = re.compile(r'(.*.jpg)\n')
+    pattern_nbbox = re.compile(r'(\d+)\n')
+    pattern_annot = re.compile(r'(\d+) (\d+) (\d+) (\d+) (\d+) '
+                               r'(\d+) (\d+) (\d+) (\d+) (\d+) \n')
+    if split == 'test':
+      annot_file = 'wider_face_split/wider_face_test_filelist.txt'
```

Comment:
Don't use static path, please use `os.path.join`

Reply:
> Don't use static path, please use `os.path.join`

switched to `os.path.join` in 8816908